src/Microsoft.VisualStudio.Threading.Analyzers/VSTHRD109AvoidAssertInAsyncMethodsAnalyzer.cs

```diff
@@ -49,43 +44,26 @@ public override void Initialize(AnalysisContext context)
             context.RegisterCompilationStartAction(ctxt =>
             {
                 var mainThreadAssertingMethods = CommonInterest.ReadMethods(ctxt.Options, CommonInterest.FileNamePatternForMethodsThatAssertMainThread, ctxt.CancellationToken).ToImmutableArray();
-                ctxt.RegisterSyntaxNodeAction(Utils.DebuggableWrapper(c => this.AnalyzeInvocation(c, mainThreadAssertingMethods)), SyntaxKind.InvocationExpression);
+                ctxt.RegisterOperationAction(Utils.DebuggableWrapper(c => this.AnalyzeInvocation(c, mainThreadAssertingMethods)), OperationKind.Invocation);
             });
         }
 
-        private void AnalyzeInvocation(SyntaxNodeAnalysisContext context, ImmutableArray<CommonInterest.QualifiedMember> mainThreadAssertingMethods)
+        private void AnalyzeInvocation(OperationAnalysisContext context, ImmutableArray<CommonInterest.QualifiedMember> mainThreadAssertingMethods)
         {
-            var functionInfo = Utils.GetContainingFunction((CSharpSyntaxNode)context.Node);
-            if (functionInfo.Function == null)
+            if (!(Utils.GetContainingFunction(context.Operation, context.ContainingSymbol) is IMethodSymbol methodSymbol))
             {
-                // One case where this happens is the use of nameof(X) in the argument of a custom attribute on a type, such as:
-                // [System.Diagnostics.DebuggerDisplay(""hi"", Name = nameof(System.Console))]
-                // class Foo { }
```

Comment:
Why did you remove the explanatory code comment?

Reply:
The comment is no longer correct. This is an `INameOfOperation`, not an invocation.